cmd/shipperctl/cmd/backup/restore.go

```diff
@@ -0,0 +1,201 @@
+		Use:   "restore",
+		Short: "Restore backup that was prepared using `shipperctl prepare` command",
+		Long: "Restore backup that was prepared using `shipperctl prepare` command" +
+			"this will update the owner reference of releases objects",
+		RunE: runRestoreCommand,
+	}
+)
+
+func init() {
+	BackupCmd.AddCommand(restoreBackupCmd)
+}
+
+func runRestoreCommand(cmd *cobra.Command, args []string) error {
+	kubeClient, err := configurator.NewKubeClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	shipperClient, err := configurator.NewShipperClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	releasesPerApplications, err := unmarshalReleasesPerApplicationFromFile()
+	if err != nil {
+		return err
+	}
+
+	confirm, err := ui.AskForConfirmation(cmd.InOrStdin(), "Would you like to see an overview of your backup?")
+	if err != nil {
+		return err
+	}
+	if confirm {
+		printReleasesPerApplications(releasesPerApplications)
+		confirm, err = ui.AskForConfirmation(cmd.InOrStdin(), "Would you like to review backup?")
+		if err != nil {
+			return err
+		}
+		if confirm {
+			data, err := marshalReleasesPerApplications(releasesPerApplications)
+			if err != nil {
+				return err
+			}
+			if _, err := cmd.OutOrStdout().Write(data); err != nil {
+				return err
+			}
+		}
+	}
+
+	confirm, err = ui.AskForConfirmation(cmd.InOrStdin(), "Would you like to restore backup?")
+	if err != nil {
+		return err
+	}
+	if !confirm {
+		return nil
+	}
+
+	if err := restore(releasesPerApplications, kubeClient, shipperClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func restore(releasesPerApplications []shipperBackupObject, kubeClient kubernetes.Interface, shipperClient versioned.Interface) error {
+	var errList []string
+	for _, obj := range releasesPerApplications {
+		// apply application
+		if err := applyApplication(kubeClient, shipperClient, obj.Application); err != nil {
+			errList = append(errList, err.Error())
+			continue
+		}
+		fmt.Printf("application %q created\n", fmt.Sprintf("%s/%s", obj.Application.Namespace, obj.Application.Name))
+
+		// get the new UID
+		uid, err := uidOfApplication(shipperClient, obj.Application.Name, obj.Application.Namespace)
+		if err != nil {
+			errList = append(errList, err.Error())
+			continue
+		}
+
+		// update owner ref for all releases and apply them
+		for _, release := range obj.Releases {
+			rel, err := updateOwnerRefUid(release, *uid)
+			if err != nil {
+				errList = append(errList, err.Error())
+				continue
+			}
+			fmt.Printf("release %q owner reference updates with uid %q\n", fmt.Sprintf("%s/%s", rel.Namespace, rel.Name), *uid)
+
+			rel.ResourceVersion = ""
+			if _, err = shipperClient.ShipperV1alpha1().Releases(rel.Namespace).Create(rel); err != nil {
+				errList = append(errList, err.Error())
+			}
+			fmt.Printf("release %q created\n", fmt.Sprintf("%s/%s", rel.Namespace, rel.Name))
```

Comment:
Do you foresee any problem if _some_ of the releases fail to be restored? (not a blocker for now, just a thought)

Reply:
Some of the releases fail to be restored == loss of history.
I don't know if this is the best way to overcome this. I think in this scenario I would want to apply the backup again, but it would fail to create the application (already there error) and would skip its releases.
I;m not sure what is the best solution here.

Reply:
What about if the "last 3" fails to restore. Would that cause problems? or just restore that particular version?

Reply:
Yeah that's way worse: if the latest release(s) fails to restore, the live version is not the latest. 
We should probably panic upon first error instead of collecting them. WDYT?

Reply:
Yes, that would neat!